cli/src/cluster_query.rs

```diff
@@ -1314,12 +1334,14 @@ pub fn process_show_block_production(
             (confirmed_blocks, start_slot)
         };
 
-    let start_slot_index = (start_slot - first_slot_in_epoch) as usize;
-    let end_slot_index = (end_slot - first_slot_in_epoch) as usize;
-    let total_slots = end_slot_index - start_slot_index + 1;
+    let start_slot_index = start_slot.saturating_sub(first_slot_in_epoch) as usize;
+    let end_slot_index = end_slot.saturating_sub(first_slot_in_epoch) as usize;
+    let total_slots = end_slot_index
+        .saturating_sub(start_slot_index)
+        .saturating_sub(1);
```

Comment:
Looks like this one should be:
```suggestion
    let total_slots = end_slot_index
        .saturating_sub(start_slot_index)
        .saturating_add(1);
```

Reply:
Thanks a lot of double checking!
Fixed.